atc/db/check_factory.go

```diff
@@ -80,14 +86,14 @@ func NewCheckFactory(
 		defaultCheckTimeout:             durations.Timeout,
 		defaultCheckInterval:            durations.Interval,
 		defaultWithWebhookCheckInterval: durations.IntervalWithWebhook,
+
+		checkBuildChan: checkBuildChan,
 	}
 }
 
-func (c *checkFactory) TryCreateCheck(ctx context.Context, checkable Checkable, resourceTypes ResourceTypes, from atc.Version, manuallyTriggered bool) (Build, bool, error) {
+func (c *checkFactory) TryCreateCheck(ctx context.Context, checkable Checkable, resourceTypes ResourceTypes, from atc.Version, manuallyTriggered bool, toDB bool) (Build, bool, error) {
```

Comment:
I wonder if we ever actually care to store check builds to the DB, even if they're manually triggered, or whether they could always just be in memory. Is there a reason you had in mind for supporting both?

Reply:
To support both, majorly for webhook checks. A resource with webhook has check interval 1-hour. If a webhook call comes to an ATC, and the ATC happens to crash, then the check will lose, meaning a the webhook call is lost, the resource can only be checked until next webhook or after 1 hour.

Whereas, periodic checks come every 10 seconds, even if an ATC crashes, lost check builds come again after 10 seconds.